File: app/src/main/java/n7/ad2/retrofit/streams/Streams.java
package n7.ad2.retrofit.streams;


public class Streams {
    private int viewers = 0;
    private Preview preview;
    private Channel channel;
    private boolean isPremium = false;

    public boolean isPremium() {
        return isPremium;
    }

    public void setPremium(boolean premium) {
        isPremium = premium;
    }

    public int getViewers() {
        return viewers;
    }

    public void setViewers(int viewers) {
        this.viewers = viewers;
    }

    public Preview getPreview() {
        return preview;
    }

    public void setPreview(Preview preview) {
        this.preview = preview;
    }

    public Channel getChannel() {
        return channel;
    }

    public void setChannel(Channel channel) {
        this.channel = channel;
    }

}

File: app/src/main/java/n7/ad2/data/source/remote/model/Streams.kt
package n7.ad2.data.source.remote.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Streams(
    @Json(name = "_total")
    val _total: Int = 0,
    @Json(name = "streams")
    val streams: List<Stream> = emptyList(),
)

class Stream(
    @Json(name = "viewers")
    val viewers: Int = 0,
    @Json(name = "preview")
    val preview: Preview? = null,
    @Json(name = "channel")
    val channel: Channel = Channel(),
    val isPremium: Boolean = false,
)

class Preview(
    @Json(name = "medium")
    val medium: String = "",
)

class Channel(
    @Json(name = "status")
    val status: String = "",
    @Json(name = "display_name")
    val name: String = "",
)
